### ISSUE TITLE
Missing parity between command line and programmatic options

### ISSUE THREAD

**Issue by sedge** [substack/node-browserify#1289]
I've been crawling the docs and I think I noticed a couple of features present in browserify, but missing from either the command line or the programmatic API.

1.  The [API](https://github.com/substack/node-browserify#browserifyfiles--opts) provides `opts.builtins` for specifying which builtins a build should include, but the command line doesn't have a way to specify this.
2.  On the other side, the [`--igv` parameter](https://github.com/substack/node-browserify#usage) allows the user to explicitly declare which globals should be defined, but the API doesn't have an equivalent.

Is this my mistake?

**Comment by jmm** [comment]
@sedge You are correct that there are some disparities between CLI and API. Sometimes it's a documentation issue, and in other cases there may be a mismatch in functionality.

I believe [`opts.insertGlobalVars`](https://github.com/substack/node-browserify#browserifyfiles--opts) is the API equivalent of `--igv`.

**Comment by sedge** [comment]
@jmm `opts.insertGlobalVars` is slightly (but significantly) different, since it passes the value directly to `insert-module-globals`. The `--igv` parameter is parsed into an object literal before being passed to that module.

**Comment by sedge** [comment]
Though, thinking about it, I guess it's close enough to work. Maybe that case is a docs issue then?

**Comment by jmm** [comment]
@sedge 
> The `--igv` parameter is parsed into an object literal before being passed to that module.

Parsed from a string into an object, yes :)

> `opts.insertGlobalVars` is slightly (but significantly) different, since it passes the value directly to `insert-module-globals` and needs to be an object.
> ...
> Though, thinking about it, I guess it's close enough to work.

That's basically par for the course. A shell CLI is very limited compared to JS and has to accept string parameters and massage them in preparation for consuming the JS API. The exact format of the input will frequently differ, as it often doesn't make sense for the API to accept string arguments where the CLI must. Same with `b.require` vs. `--require|-r` for example: you can `-r some:thing` to alias `some` to `thing`, but you can't `b.require('some:thing')`.

> Maybe that case is a docs issue then?

browserify's `insertGlobalVars` is already documented as being used as the `insert-module-globals` `opts.vars` value, but the documentation over there could be better.

**Comment by sedge** [comment]
@jmm Thanks for taking the time to clarify! I hadn't considered the inherent limitations of the command line.

**Comment by jmm** [comment]
@sedge No problem. If you can write JS and you're doing anything repetitive with browserify, I highly recommend just writing scripts against the API and eschewing the CLI unless you really need it.

**Comment by mattdesl** [comment]
Something I have done in the past is use the `require('browserify/bin/args')`. This is mainly only useful if you need to take the user `process.argv` and pass it directly to browserify (i.e. for subarg syntax to work nicely). You can specify API overrides like so:

https://github.com/Jam3/chromeo/blob/e3aadb3b09cdb6a3322f7ea25374f8767140c9ac/lib/create-browserify.js#L4-L13

**Comment by yeya** [comment]
Another missing option (that after a lot of search cause me to come here) is `--bare` and `--node` commands
